### PR TITLE
Add NSP API to workspaces

### PR DESCRIPTION
**Pull request by akshitgoyal**
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

> [!NOTE]
> As of January 2024 there is no PR assignee. This is expected. See https://aka.ms/azsdk/pr-arm-review.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

diagram

<details>
<summary>Click here to see the details of Step 1, Breaking Changes review</summary>

If you are in purview of Step 1 of the diagram, follow the [Breaking Changes review process].  
**IMPORTANT!** This applies even if you believe your PR was mislabeled, for any reason, including tool failure.
</details>

<details> 
<summary>Click here to see the details of Step 2, ARM review</summary>

See https://aka.ms/azsdk/pr-arm-review.
</details>

<details>
<summary> Click here to see the diagram footnotes</summary>

### Diagram footnotes

[1] See [ARM review queue] (for **PR merge** queues, see [2]).  
[2] [public repo merge queue], [private repo merge queue] (for **ARM review** queue, [1])  
The ARM reviewer on-call engineer visits the merge queue twice a day, so the approximate ETA for merges is 12 - 24 hours.

</details>

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [x] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been generated using [OpenAPI Hub](https://aka.ms/openapihub)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with PR workflow diagram Step 2 (ARM review), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

[ARM review queue]: https://aka.ms/azsdk/pr-arm-review#about-the-arm-review-queue
[public repo merge queue]: https://github.com/Azure/azure-rest-api-specs/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+draft%3Afalse+sort%3Acreated-asc
[private repo merge queue]: https://github.com/Azure/azure-rest-api-specs-pr/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+-label%3AApproved-OkToMerge+draft%3Afalse+sort%3Acreated-asc
[Breaking Changes review process]: https://aka.ms/brch

### CODE REVIEW

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json
Comment (ramoka178) [inline specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json:375]: >severity [](http://example.com/codeflow?start=9&length=8)

enum ?

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json
Comment (ramoka178) [inline specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json:372]: >issueType [](http://example.com/codeflow?start=9&length=9)

enum ?

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json
Comment (ramoka178) [inline specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json:66]: >BadRequest [](http://example.com/codeflow?start=28&length=10)

generic description to suit all errors?

here and across the PR

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json
Comment (ramoka178) [inline specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json:154]: Consider adding retry-after ?

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json
Comment (ramoka178) [inline specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json:219]: >provisioningState [](http://example.com/codeflow?start=9&length=17)

make it enum

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json
Comment (ramoka178) [inline specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json:245]: >name [](http://example.com/codeflow?start=9&length=4)

any patterns, restrictions on the name ?

here and across PR

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json
Comment (ramoka178) [inline specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json:258]: >direction [](http://example.com/codeflow?start=9&length=9)

can be from a fixed set of enums ?

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json
Comment (ramoka178) [inline specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json:267]: >fullyQualifiedDomainNames [](http://example.com/codeflow?start=9&length=25)

arm-ids ?

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json
Comment (ramoka178) [inline specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json:294]: >accessMode [](http://example.com/codeflow?start=9&length=10)

enum ?

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json
Comment (ramoka178) [inline specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json:303]: >id [](http://example.com/codeflow?start=9&length=2)

arm-id ?

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json
Comment (ramoka178) [inline specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json:308]: name says guid, but type is string. 

please make it consistent.

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json
Comment (ramoka178) [inline specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json:318]: >NSPConfigPerimeter [](http://example.com/codeflow?start=5&length=18)

same as NSPConfigNetworkSecurityPerimeterRule. can be merged?

Comment (akshitgoyal) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json:334]: merged

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json
Comment (akshitgoyal) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json:319]: resovled

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json
Comment (akshitgoyal) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json:270]: resolved

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json
Comment (akshitgoyal) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json:257]: cannot enforce restrictions as these models are provided by networking team and need to be consistent across services

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json
Comment (akshitgoyal) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json:151]: we always send 202

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json
Comment (akshitgoyal) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json:154]: we always send 202

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json
Comment (akshitgoyal) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json:389]: resolved

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json
Comment (akshitgoyal) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json:401]: resolved

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json
Comment (akshitgoyal) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json]: resolved

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json
Comment (akshitgoyal) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json]: resolved

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json
Comment (ramoka178) [inline specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json:344]: My comment was showing up in another commit. 

name says guid, but type is string. can it be perimeterId ?
please make it consistent.

Comment (akshitgoyal) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json:344]: we cannot change the names of the fields since these models need to be same across all services. For example, the storage service has this as well which has gone through arm review already:
https://github.com/Azure/azure-rest-api-specs/blob/cf230c94500f8ee044acc68cb7bab37eace5671d/specification/storage/resource-manager/Microsoft.Storage/stable/2023-04-01/networkSecurityPerimeter.json#L421

Comment (akshitgoyal) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json:344]: I couldn't find examples of any open api specs that have guid types in the repo, can you point me to an example?

**Comment by akshitgoyal** [comment]
@ramoka178 can we get a signoff on this PR?

**Comment by TimLovellSmith** [comment]
Just a heads up. I need to go through some careful review of these NSP definitions, I'm trying to figure out how to avoid many inconsistent definitions of these across services.

**Comment by TimLovellSmith** [comment]
@akshitgoyal I'm working on adding NetworkSecurityPerimeters to common types, as something like this is clearly urgently needed! Please help me review my PR.

https://github.com/Azure/azure-rest-api-specs/pull/28958

**Comment by TimLovellSmith** [comment]
I think you're nearly all good at this point, except for finalizing NSP type definitions and fixing SDK errors.
I'm okay to sign off an ARM manifest PR at this point if you have one this is blocked on, because of the new types - no changes are needed in type names, that I can see.

**Comment by akshitgoyal** [comment]
sounds good! thanks @TimLovellSmith

**Review by TimLovellSmith — CHANGES_REQUESTED** [review]
Marking this PR as need to refactor to refer to NSP definitions from

https://github.com/Azure/azure-rest-api-specs/pull/28958

before merging.

To be clear the intent of this feedback is not to say that ARM is not signed off (ARM is signed off), nor is it to block manifest deployments (they should be unblocked)

**Comment by akshitgoyal** [comment]
> Marking this PR as need to refactor to refer to NSP definitions from
> 
> #28958
> 
> before merging.
> 
> To be clear the intent of this feedback is not to say that ARM is not signed off (ARM is signed off), nor is it to block manifest deployments (they should be unblocked)

@TimLovellSmith Can we receive the ArmSignOff label. I will refactor the code as soon as your PR is merged. Thanks!

**Comment by Alancere** [comment]
Hi @akshitgoyal, please fix the `SDK azure-sdk-for-go` CI issue

**Comment by TimLovellSmith** [comment]
@akshitgoyal Please refactor by using references to the [new] common types NSP definitions, letting me know of any blockers

https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v5/networksecurityperimeter.json

**Comment by akshitgoyal** [comment]
> @akshitgoyal Please refactor by using references to the [new] common types NSP definitions, letting me know of any blockers
> 
> https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v5/networksecurityperimeter.json

I have updated the PR with new common NSP specs.

**Comment by akshitgoyal** [comment]
@TimLovellSmith seems like NetworkSecurityPerimeterConfigurationNameParameter is causing issues. Can you take a look?

**Comment by TimLovellSmith** [comment]
@akshitgoyal You can add a suppression, or copy/paste the parameter definition and define your own pattern as reusing the parameter definition isn't critical, its just a convenience.

**Comment by TimLovellSmith** [comment]
@akshitgoyal  Actually, is there a casing issue in your operation paths?

It should be ''/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.OperationalInsights/workspaces/{workspaceName}/networkSecurityPerimeterConfigurations/{networkSecurityPerimeterConfigurationNameParameter}''

with a little 'n' at the start of the the parameter name (camel case), instead of a big 'N' (pascal case).

**Comment by akshitgoyal** [comment]
> Hi @akshitgoyal, please fix the `SDK azure-sdk-for-go` CI issue

Hey @Alancere, can you confirm if these errors are caused by current new changes? Can you help us resolve these?

**Comment by akshitgoyal** [comment]
@TimLovellSmith I had to add parameter definition for `NetworkSecurityPerimeterConfigurationNameParameter` because the current common types don't have a "pattern" restriction and the CI checks are failing because of that.  I think the common types definition needs to be updated to account for this. LintDiff will fail without that. For context, you can see this previous run: https://github.com/Azure/azure-rest-api-specs/pull/28733/checks?check_run_id=25582443380

if these changes look okay, can you approve them, so we can get this merged? Thanks

**Comment by Alancere** [comment]
> > Hi @akshitgoyal, please fix the `SDK azure-sdk-for-go` CI issue
> 
> Hey @Alancere, can you confirm if these CI issues are caused by current new changes? Can you help us resolve these?

The possible reason of Duplicate Schema is that different common-type versions are used. I find that `package-2024-04` contains v2 and v5

For example:
`ErrorResponse`
- https://github.com/Azure/azure-rest-api-specs/blob/akshitgoyal/microsoft-operationinsights-nsp/specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2020-08-01/OperationStatuses.json#L110
- https://github.com/Azure/azure-rest-api-specs/blob/akshitgoyal/microsoft-operationinsights-nsp/specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json#L68

**Comment by akshitgoyal** [comment]
> > > Hi @akshitgoyal, please fix the `SDK azure-sdk-for-go` CI issue
> > 
> > 
> > Hey @Alancere, can you confirm if these CI issues are caused by current new changes? Can you help us resolve these?
> 
> The possible reason of `Duplicate Schema` is that different common-type versions are used. I find that `package-2024-04` contains v2 and v5
> 
> For example: `ErrorResponse`
> 
> * https://github.com/Azure/azure-rest-api-specs/blob/akshitgoyal/microsoft-operationinsights-nsp/specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2020-08-01/OperationStatuses.json#L110
> * https://github.com/Azure/azure-rest-api-specs/blob/akshitgoyal/microsoft-operationinsights-nsp/specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json#L68

@Alancere Thanks for looking into this. My PR only makes changes to this file:
* https://github.com/Azure/azure-rest-api-specs/blob/akshitgoyal/microsoft-operationinsights-nsp/specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json

For resources like `ProxyResource`, `ErrorDetail`, etc, I cannot change them because these new APIs and this PR doesn't reference them directly. Is there a way to suppress these?

**Comment by Alancere** [comment]
> > > > Hi @akshitgoyal, please fix the `SDK azure-sdk-for-go` CI issue
> > > 
> > > 
> > > Hey @Alancere, can you confirm if these CI issues are caused by current new changes? Can you help us resolve these?
> > 
> > 
> > The possible reason of `Duplicate Schema` is that different common-type versions are used. I find that `package-2024-04` contains v2 and v5
> > For example: `ErrorResponse`
> > 
> > * https://github.com/Azure/azure-rest-api-specs/blob/akshitgoyal/microsoft-operationinsights-nsp/specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2020-08-01/OperationStatuses.json#L110
> > * https://github.com/Azure/azure-rest-api-specs/blob/akshitgoyal/microsoft-operationinsights-nsp/specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json#L68
> 
> @Alancere Thanks for looking into this. My PR only makes changes to this file:
> 
> * https://github.com/Azure/azure-rest-api-specs/blob/akshitgoyal/microsoft-operationinsights-nsp/specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json
> 
> For resources like `ProxyResource`, `ErrorDetail`, etc, I cannot change the version for them because these new APIs and this PR doesn't reference them directly. Is it possible these were introduced because of a different PR in past? Is there a way to suppress these?

Some are indirectly referenced, and it is recommended to unify the commontype version

**Comment by akshitgoyal** [comment]
> > > > > Hi @akshitgoyal, please fix the `SDK azure-sdk-for-go` CI issue
> > > > 
> > > > 
> > > > Hey @Alancere, can you confirm if these CI issues are caused by current new changes? Can you help us resolve these?
> > > 
> > > 
> > > The possible reason of `Duplicate Schema` is that different common-type versions are used. I find that `package-2024-04` contains v2 and v5
> > > For example: `ErrorResponse`
> > > 
> > > * https://github.com/Azure/azure-rest-api-specs/blob/akshitgoyal/microsoft-operationinsights-nsp/specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2020-08-01/OperationStatuses.json#L110
> > > * https://github.com/Azure/azure-rest-api-specs/blob/akshitgoyal/microsoft-operationinsights-nsp/specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json#L68
> > 
> > 
> > @Alancere Thanks for looking into this. My PR only makes changes to this file:
> > 
> > * https://github.com/Azure/azure-rest-api-specs/blob/akshitgoyal/microsoft-operationinsights-nsp/specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json
> > 
> > For resources like `ProxyResource`, `ErrorDetail`, etc, I cannot change the version for them because these new APIs and this PR doesn't reference them directly. Is it possible these were introduced because of a different PR in past? Is there a way to suppress these?
> 
> Some are indirectly referenced, and it is recommended to unify the common-type version

@Alancere I have fixed most of them. However, for `Resource` and `ProxyResource`, these are referenced in `../../../../../common-types/resource-management/v5/networksecurityperimeter.json`. `networksecurityperimeter.json` is only available in v5 and is necessary for this PR.

**Comment by raych1** [comment]
@akshitgoyal , you will need to copy the models you referenced from `networksecurityperimeter.json` to your own swagger then update the `Resource` and `ProxyResource` reference from V2 common types if you cannot unify other version swaggers to reference V5 types.

**Comment by akshitgoyal** [comment]
> @akshitgoyal , you will need to copy the models you referenced from `networksecurityperimeter.json` to your own swagger then update the `Resource` and `ProxyResource` reference from V2 common types if you cannot unify other version swaggers to reference V5 types.

@raych1 @Alancere all the issues and types are fixed. Can we get a signoff for sdk?

**Review by TimLovellSmith — APPROVED** [review]

**Comment by akshitgoyal** [comment]
@MaryGao @qiaozha @kazrael2119 can we get the `BreakingChange-JavaScript-Sdk-Approved` label?

**Comment by kazrael2119** [comment]
block this pr as https://github.com/Azure/azure-rest-api-specs/pull/28945#issuecomment-2137982445

**Review by TimLovellSmith — APPROVED** [review]

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/networksecurityperimeter.json
Comment (TimLovellSmith) [inline specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/networksecurityperimeter.json:10]: To avoid future maintainers modifying these definitions, will you be doing another update to consume common-types v5, and remove this reference, in e.g. next api version?

**Comment by akshitgoyal** [comment]
> block this pr as [#28945 (comment)](https://github.com/Azure/azure-rest-api-specs/pull/28945#issuecomment-2137982445)

@kazrael2119 I went through the PR you mentioned. Our package only includes the stable version of APIs. We haven't included any preview versions. Please let me know what is blocking our PR and how can we get unblocked on this. Thanks!

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/networksecurityperimeter.json
Comment (akshitgoyal) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/networksecurityperimeter.json:10]: I'll make the team aware of this so that this can be prevented/fixed in future api versions. Thanks Tim!

**Comment by kazrael2119** [comment]
> > block this pr as [#28945 (comment)](https://github.com/Azure/azure-rest-api-specs/pull/28945#issuecomment-2137982445)
> 
> @kazrael2119 I went through the PR you mentioned. Our package only includes the stable version of APIs. We haven't included any preview versions. Please let me know what is blocking our PR and how can we get unblocked on this. Thanks!

@akshitgoyal the block is Operational.Insight has multi different api versions in one tag and this need @JeffreyRichter approve
in that methioned pr, It not only has mixed preview and stable version issue , but also has multi different api versions issue

**Comment by JeffreyRichter** [comment]
I had a meeting with @dulikvor yesterday and his team is going to split the swaggers into separate services (fixing some versioning) and make a new PR.

**Comment by akshitgoyal** [comment]
@JeffreyRichter @kazrael2119 Thank you so much for the context. Are there any actions on us? do we have to do this PR again to account for any new changes? Also if possible, can you please tell me an estimate of delay to this PR so I can let teams concerned with this PR know?

**Comment by JeffreyRichter** [comment]
You need to discuss with @dulikvor

**Pull request closed without merging**
